openpype/plugins/publish/extract_color_transcode.py

```diff
@@ -0,0 +1,214 @@
+
+            for _, output_def in profile.get("outputs", {}).items():
+                new_repre = copy.deepcopy(repre)
+
+                original_staging_dir = new_repre["stagingDir"]
+                new_staging_dir = get_transcode_temp_directory()
+                new_repre["stagingDir"] = new_staging_dir
+                files_to_convert = new_repre["files"]
+                if not isinstance(files_to_convert, list):
+                    files_to_convert = [files_to_convert]
+
+                files_to_delete = copy.deepcopy(files_to_convert)
+
+                output_extension = output_def["output_extension"]
+                output_extension = output_extension.replace('.', '')
+                if output_extension:
+                    if new_repre["name"] == new_repre["ext"]:
+                        new_repre["name"] = output_extension
+                    new_repre["ext"] = output_extension
+
+                target_colorspace = output_def["output_colorspace"]
+                if not target_colorspace:
+                    raise RuntimeError("Target colorspace must be set")
+
+                for file_name in files_to_convert:
+                    input_filepath = os.path.join(original_staging_dir,
+                                                  file_name)
+                    output_path = self._get_output_file_path(input_filepath,
+                                                             new_staging_dir,
+                                                             output_extension)
+                    convert_colorspace(
+                        input_filepath,
+                        output_path,
+                        config_path,
+                        source_colorspace,
+                        target_colorspace,
+                        self.log
+                    )
+
+                instance.context.data["cleanupFullPaths"].extend(
+                    files_to_delete)
+
+                custom_tags = output_def.get("custom_tags")
+                if custom_tags:
+                    if not new_repre.get("custom_tags"):
+                        new_repre["custom_tags"] = []
+                    new_repre["custom_tags"].extend(custom_tags)
+
+                # Add additional tags from output definition to representation
+                for tag in output_def["tags"]:
+                    if tag not in new_repre["tags"]:
+                        new_repre["tags"].append(tag)
+
+                instance.data["representations"].append(new_repre)
+
+    def _get_output_file_path(self, input_filepath, output_dir,
+                              output_extension):
+        """Create output file name path."""
+        file_name = os.path.basename(input_filepath)
+        file_name, input_extension = os.path.splitext(file_name)
+        if not output_extension:
+            output_extension = input_extension
+        new_file_name = '{}.{}'.format(file_name,
```

Comment:
```suggestion
        new_file_name = '{}{}'.format(file_name,
```
Getting double `.` here from the format and `output_extension`.

Reply:
This has not been implemented.